### ISSUE TITLE
Allow setting default max-workers and max-num-split values

### ISSUE THREAD

**Issue by displacedaussie** [Netflix/metaflow#490]
The default values for `--max-workers` (16) and `--max-num-splits` (100) are fine for many use cases, but for some workloads and metaflow environments, it would be useful if it were possible to set new default values. Currently we rely on the user to override them on the command line for each run.

We have some Flows where we always want max-workers to be 32 (or 64), and max-num-splits to be 256.

Reading `METAFLOW_DEFAULT_MAX_WORKERS` and `METAFLOW_DEFAULT_MAX_NUM_SPLITS` from the environment would allow us to set these in the metaflow profile/config file and then not have to worry about them being set on the command line.

**Comment by savingoyal** [comment]
@displacedaussie Thanks for opening the issue. Thanks to `click`, this functionality already exists. You can set the environment variables - `METAFLOW_RUN_MAX_WORKERS` and `METAFLOW_RUN_MAX_NUM_SPLITS` to achieve the desired outcome 😄 

```
from metaflow import FlowSpec, step

class ForeachFlow(FlowSpec):

    @step
    def start(self):
        self.titles = ['Stranger Things',
                       'House of Cards',
                       'Narcos']
        self.next(self.a, foreach='titles')

    @step
    def a(self):
        self.title = '%s processed' % self.input
        self.next(self.join)

    @step
    def join(self, inputs):
        self.results = [input.title for input in inputs]
        self.next(self.end)

    @step
    def end(self):
        print('\n'.join(self.results))

if __name__ == '__main__':
    ForeachFlow()
```

```
METAFLOW_RUN_MAX_NUM_SPLITS=1 python3 ForEach.py run
```
^ this command will fail - 
```
Step start (task-id 1) failed: Foreach in step start yielded 3 child steps which is more than the current maximum of 1 children. You can raise the maximum with the --max-num-splits option.
```

```
METAFLOW_RUN_MAX_NUM_SPLITS=1 python3 ForEach.py run --max-num-splits 3
```
^ while this will succeed

**Comment by displacedaussie** [comment]
Thank you Savin, that is definitely useful. I hadn't realised click could do this.

I guess there are two downsides to this though:

1) We need separate environment variables for `run` and `resume`, which means setting the same value in multiple places:

```
MEATFLOW_RUN_MAX_WORKERS=32
MEATFLOW_RESUME_MAX_WORKERS=32
METAFLOW_RUN_MAX_NUM_SPLITS=256
METAFLOW_RESUME_MAX_NUM_SPLITS=256
```

2) We can't put it in the metaflow configuration file. Being able to share a single metaflow configuration file is extremely useful!

**Comment by savingoyal** [comment]
That's true. We have been thinking about a better configuration management system for both Metaflow and Flows. There is a ticket open #430 to that end with the team at Zillow. While we ideate through the options for a longer-term solution, a short-term option is to export the entire metaflow configuration as well as these parameters (`METAFLOW_RUN_MAX_WORKERS` etc.) in your `bashrc` file.
